middlewares/validators/progresses.js

```diff
@@ -1,7 +1,14 @@
 const joi = require("joi");
-const { VALID_PROGRESS_TYPES } = require("../../constants/progresses");
+const { VALID_PROGRESS_TYPES, RESPONSE_MESSAGES } = require("../../constants/progresses");
 
 const validateCreateProgressRecords = async (req, res, next) => {
+  // Check if the request is being made on a Sunday
+  const today = new Date();
+  if (today.getDay() === 0) {
+    // Sunday corresponds to day index 0
+    res.boom.badRequest(RESPONSE_MESSAGES.PROGRESS_DOCUMENT_NON_WORKING_DAYS);
+  }
+
```

Comment:
there is a catch here @khalatevarun . Actually on Sundays too until 6 am IST  we will allow users to update. As for the current day users can submit their standup up till next day 6 am. Similarly someone can send sunday standup on Monday 5 am. This logic will miss both these cases.

Reply:
got it, will add support to these two cases